### PR TITLE
GroupedList: Fix bug where selection would not work when keyboarding off a header

### PR DESCRIPTION
**Pull request by aaronyzhou**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file if publishing 
- [X] New feature, bugfix, or enhancement
  - [ ] Includes tests
- [ ] Documentation update

#### Description of changes

Fix selection and focusing when using arrow keys to navigate on group details list.
```
G1
 a
 b
G2
 x
 y
```
Using the arrow keys to navigate from x -> b will now do selection and focus properly.

#### Focus areas to test

Grouped list and all other places using the selection zone.

### CODE REVIEW

**Comment by micahgodbolt** [comment]
@aaronyzhou can you run `$ rush change` and submit a change file?

**Comment by aaronyzhou** [comment]
I had to manually create a rush change file because the command didn't run for some reason.

**Comment by dzearing** [comment]
@aaronyzhou Can you explain the behavior you're changing? From reading the source, it seems that if you set focus to something that isn't a row, we clear selection. That doesn't seem quite right. What is the behavior you're trying to achieve exactly?

**Comment by aaronyzhou** [comment]
This is the current behavior when using the up/down keys to navigate.
groupdetailslisterror

My fix changes it to behave like this
groupdetailslistfixed

**Comment by micahgodbolt** [comment]
A+ gif game.

**Comment by micahgodbolt** [comment]
Ah, moving from the header to an item doesn't select the item, right? It's kinda hard to see the difference, but that certainly is a nasty bug.

**Comment by micahgodbolt** [comment]
group

File: packages/office-ui-fabric-react/src/utilities/selection/SelectionZone.tsx [hunk 1]
```diff
@@ -139,6 +139,8 @@ export class SelectionZone extends BaseComponent<ISelectionZoneProps, {}> {
         } else {
           this._onItemSurfaceClick(ev, index);
         }
+      } else {
+        selection.setAllSelected(false);
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/utilities/selection/SelectionZone.tsx]: This line I think is unnecessary; we don't need to clear selection when focus is set on a header.

File: packages/office-ui-fabric-react/src/utilities/selection/SelectionZone.tsx [hunk 2]
```diff
@@ -337,10 +339,10 @@ export class SelectionZone extends BaseComponent<ISelectionZoneProps, {}> {
 
         target = getParent(target);
       }
-
-      // A key was pressed while an item in this zone was focused.
-      this._shouldHandleFocus = true;
     }
+
+    // A key was pressed while an item in this zone was focused.
+    this._shouldHandleFocus = true;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/utilities/selection/SelectionZone.tsx]: I think this move here might be all you need!

File: packages/office-ui-fabric-react/src/utilities/selection/SelectionZone.tsx (hunk 1 above)
Comment (aaronyzhou) [inline packages/office-ui-fabric-react/src/utilities/selection/SelectionZone.tsx]: Without this line, if you up into a header, the selection stays on the previous row. But its better if you're doing multi select. 
groupdetailslist2

Comment (dzearing) [inline packages/office-ui-fabric-react/src/utilities/selection/SelectionZone.tsx]: Right. That's the design. Moving to a non selectable entity should not be clearing your selection.

Comment (aaronyzhou) [inline packages/office-ui-fabric-react/src/utilities/selection/SelectionZone.tsx]: Updated.

**Pull request merged**